common/client/transaction_sender_test.go

```diff
@@ -357,11 +386,13 @@ func TestTransactionSender_SendTransaction_aggregateTxResults(t *testing.T) {
 		}
 
 		t.Run(testCase.Name, func(t *testing.T) {
-			_, txResult, err := aggregateTxResults(testCase.ResultsByCode)
-			if testCase.ExpectedTxResult == "" {
-				assert.NoError(t, err)
-			} else {
-				assert.EqualError(t, txResult, testCase.ExpectedTxResult)
+			txResult, err := aggregateTxResults(testCase.ResultsByCode)
+			if !testCase.ExpectedNilResult {
```

Comment:
Why do we need this field? Can't we just verify that txResult.TxError matches ExpectedTxResult, if `ExpectedTxResult` is not empty?